google/cloud/storage/internal/grpc_hmac_key_request_parser_test.cc

```diff
@@ -113,6 +117,85 @@ TEST(GrpcBucketRequestParser, GetHmacKeyRequestAllOptions) {
   EXPECT_THAT(actual, IsProtoEqual(expected));
 }
 
+TEST(GrpcBucketRequestParser, ListHmacKeysRequestAllOptions) {
+  v2::ListHmacKeysRequest expected;
+  ASSERT_TRUE(google::protobuf::TextFormat::ParseFromString(
+      R"pb(
+        project: "projects/test-project-id"
+        page_size: 42
+        page_token: "test-page-token"
+        service_account_email: "test-service-account-email"
+        show_deleted_keys: true
+        common_request_params: { user_project: "test-user-project" }
+      )pb",
+      &expected));
+
+  ListHmacKeysRequest req("test-project-id");
+  req.set_page_token("test-page-token");
+  req.set_multiple_options(Deleted(true), MaxResults(42),
+                           ServiceAccountFilter("test-service-account-email"),
+                           UserProject("test-user-project"));
+
+  auto const actual = GrpcHmacKeyRequestParser::ToProto(req);
+  EXPECT_THAT(actual, IsProtoEqual(expected));
+}
+
+TEST(GrpcBucketRequestParser, ListHmacKeysResponseFull) {
+  v2::ListHmacKeysResponse input;
+  ASSERT_TRUE(google::protobuf::TextFormat::ParseFromString(
+      R"pb(
+        hmac_keys {
+          id: "test-id-1"
+          access_id: "test-access-id-1"
+          project: "projects/test-project-id"
+          service_account_email: "test-service-account-email"
+          state: "ACTIVE"
+          create_time { seconds: 1652099696 nanos: 789000000 }
+          update_time { seconds: 1652186096 nanos: 789000000 }
+        }
+        hmac_keys {
+          id: "test-id-2"
+          access_id: "test-access-id-2"
+          project: "projects/test-project-id"
+          service_account_email: "test-service-account-email"
+          state: "ACTIVE"
+          create_time { seconds: 1652099696 nanos: 789000000 }
+          update_time { seconds: 1652186096 nanos: 789000000 }
+        }
+        next_page_token: "test-next-page-token"
+      )pb",
+      &input));
+
+  auto actual = GrpcHmacKeyRequestParser::FromProto(input);
+  // Obtained the magic base64 string using:
+  //     /bin/echo  -n "0123456789" | openssl base64 -e
```

Comment:
nit: remove artifact

Reply:
Fixed I hope.

Reply:
Sorry, I meant delete (what I assumed to be a copy-paste artifact) from here: https://github.com/googleapis/google-cloud-cpp/blob/bbaeac0c1cfbf96f2cf1718949c70d7604067af7/google/cloud/storage/internal/grpc_hmac_key_request_parser_test.cc#L68-L70

I don't see a secret. should there be one?

Reply:
I see now, fixed.